### PR TITLE
Upgrade to .NET 8

### PR DESCRIPTION
**Pull request by flcl42**

### CODE REVIEW

**Review by rubo — COMMENTED** [review]
Hold on with merging, please.

**Review by rubo — COMMENTED** [review]
There are so many warnings mostly about nullable annotations. Would be great to resolve them. If not in this PR, maybe separately.

File: src/cs-multihash
[the remark this answers is not in the document]
Comment (flcl42) [reply src/cs-multihash:1]: Good catch, I'll update submodules

[the remark this answers is not in the document]
Comment (flcl42) [reply src/cs-multihash:1]: Done, and it's displays the same

**Comment by flcl42** [comment]
> There are so many warnings mostly about nullable annotations. Would be great to resolve them. If not in this PR, maybe separately.

https://github.com/NethermindEth/dotnet-libp2p/issues/63 let's fix it later

**Review by rubo — APPROVED** [review]

**Pull request merged**